From: Jeffrey Altman <jaltman@secure-endpoints.com>
Date: Wed, 27 May 2020 02:35:46 -0400
Subject: krb5: not_found must free krb5_get_error_message string

Even though krb5_get_error_message() returns 'const char *' the
C-string is allocated and must be freed using krb5_free_error_message().

Change-Id: I8d4ef6fce12f113617443d15abadf51f1e04cf1a
---
 lib/krb5/get_cred.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lib/krb5/get_cred.c b/lib/krb5/get_cred.c
index 70b3e5f..a9c8b18 100644
--- a/lib/krb5/get_cred.c
+++ b/lib/krb5/get_cred.c
@@ -683,16 +683,17 @@ static int
 not_found(krb5_context context, krb5_const_principal p, krb5_error_code code)
 {
     krb5_error_code ret;
-    const char *err;
     char *str;
+    const char *err;
 
-    err = krb5_get_error_message(context, code);
     ret = krb5_unparse_name(context, p, &str);
     if(ret) {
 	krb5_clear_error_message(context);
 	return code;
     }
+    err = krb5_get_error_message(context, code);
     krb5_set_error_message(context, code, N_("%s (%s)", ""), err, str);
+    krb5_free_error_message(context, err);
     free(str);
     return code;
 }
